docs/DeveloperGuide.adoc

```diff
@@ -767,6 +785,255 @@ Use case ends.
+
+[none]
+* 1a. The given index is invalid.
++
+[none]
+** 1a1. AddressBook shows an error message.
++
+Use case ends.
+
+[none]
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+[none]
+* 5a. User cancels deletion.
++
+Use case ends.
+
+[discrete]
+=== Use case: Flexible contact addition
+
+*MSS*
+
+1.  User requests to add a person
+2.  User inputs desired details, eg: address, email, birthday, etc...
+3.  AddressBook adds the person to the list
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The user input is invalid.
++
+[none]
+** 2a1.  AddressBook shows an error message.
++
+Use case ends.
+
+[discrete]
+=== Use case: Copy email address
+
+*MSS*
+
+1.  User requests to list persons
+2.  AddressBook shows a list of persons
+3.  User requests to copy a specific person’s email address to clipboard
+4.  AddressBook copy the person’s email address to clipboard
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+* 3a. The specified person’s email address is empty.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: Favourite contacts
+
+*MSS*
+
+1.  User requests to list commonly contacted persons
+2.  AddressBook saves and displays list without command
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. The list is empty.
++
+Use case ends.
+
+[discrete]
+=== Use case: Highlight contacts with same name
+
+*MSS*
+
+1.  User requests to add a person
+2.  AddressBook notify if name entered matches any name in list
+3.  User edit the name of contact
+4.  AddressBook removes the notification
++
+Use case ends.
+
+[discrete]
+=== Use case: Highlight invalid command
+
+*MSS*
+
+1.  User inputs command
+2.  AddressBook highlights invalid command
+3.  User edits command
+4.  Addressbook removes highlight
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. Command entered is valid.
++
+Use case ends.
+
+[discrete]
+=== Use case: Add photos to contacts
+
+*MSS*
+
+1.  User requests to list persons
+2.  AddressBook shows a list of persons
+3.  User picks a person
+4.  User requests to add a photo to contacts
+5.  AddressBook adds photo to contact
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+* 4a. Contact already has a photo.
++
+[none]
+** 4a1. AddressBook displays message “already has a photo”.
++
+** 4a2. AddressBook displays a dialogue window asking user if user wants to change the current photo.
++
+Use case resumes at step 5.
+
+[discrete]
+=== Use case: Password feature
+
+*MSS*
+
+1.  User runs the application
+2.  AddressBook displays a window asking user to input password
+3.  User inputs password
+4.  AddressBook displays main window
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 3a. User inputs wrong password.
++
+[none]
+** 3a1. AddressBook displays message “Wrong password”.
++
+Use case resumes at step 2.
+
```

Comment:
Overall, it's good that you write use cases for each user stories, but some of them shouldn't be a user stories and also some use cases are wrong. Please check them again and update it accordingly.